v4l2rtspserver-master/src/ImpEncoder.cpp

```diff
@@ -665,16 +693,20 @@ static void* update_thread(void *p) {
             firstConfigPass = false;
         }
 
-        // Read the current time
-        struct timespec spec;
-        clock_gettime(CLOCK_REALTIME, &spec);
-
-        // Sleep until just a little after the next second
-        spec.tv_sec = 0;
-        spec.tv_nsec = 101000000L - spec.tv_nsec;
+        if (newConfig->osdTime == 0)
+        {
+            // Read the current time
+            struct timespec spec;
+            clock_gettime(CLOCK_REALTIME, &spec);
 
-        nanosleep(&spec, NULL);
+            // Sleep until just a little after the next second
+            spec.tv_sec = 0;
+            spec.tv_nsec = 101000000L - spec.tv_nsec;
 
+            nanosleep(&spec, NULL);
+        } else {
+            sleep(newConfig->osdTime);
```

Comment:
Having `newConfig->osdTime` default to `1` and then doing something like this would work better I think, since using `sleep` re-introduces the old problem of the OSD timestamp counting unevenly:

    spec.tv_nsec = (newConfig->osdTime * 100000000L + 1000000L) - spec.tv_nsec;